File: backend2/models/necklace_model.js
// models/necklace.js
const mongoose = require("mongoose");

const necklaceSchema = new mongoose.Schema({
  idNecklace: { type: String, required: true },
  data: [
    {
      time: { type: Number, required: true },
      acc: { type: Number, required: true },
      gyr: { type: Number, required: true },
      temp: { type: Number, required: true },
      pulse: { type: Number, required: true },
    },
  ],
});

module.exports = mongoose.model("necklaceData", necklaceSchema);

File: frontend/lib/models/necklace_model.dart
class NecklaceDataPoint {
  final int time;
  final double acc;
  final double gyr;
  final double temp;
  final double pulse;

  NecklaceDataPoint({
    required this.time,
    required this.acc,
    required this.gyr,
    required this.temp,
    required this.pulse,
  });

  factory NecklaceDataPoint.fromJson(Map<String, dynamic> json) {
    return NecklaceDataPoint(
      time: json['time'],
      acc: json['acc'],
      gyr: json['gyr'],
      temp: json['temp'],
      pulse: json['pulse'],
    );
  }
}
